runtime/jcl/common/thread.cpp

```diff
@@ -417,22 +417,32 @@ Java_java_lang_Thread_currentCarrierThread(JNIEnv *env, jclass clazz)
 void JNICALL
 Java_java_lang_Thread_setCurrentThread(JNIEnv *env, jclass clazz, jobject thread)
 {
-	Assert_JCL_unimplemented();
+	J9VMThread *currentThread = (J9VMThread*)env;
+	VM_VMAccess::inlineEnterVMFromJNI(currentThread);
+	currentThread->threadObject = J9_JNI_UNWRAP_REFERENCE(thread);
```

Comment:
you need to use barrier APIs for this

Reply:
Assuming `issueWriteBarrier()` is needed here, wondering what's the use scenario/concern requiring this perf sensitive call?

Reply:
I meant GC barrier, ie J9STATIC_OBJECT_STORE, J9STATIC_OBJECT_LOAD

Reply:
Storing into or reading from references in a native structure that acts as a _strong_ root (such as J9VMThread, thread slots etc), does not require object assess barrier. 
It would be nice to use it, for potential tracking/debugging, and since in theory there are GCs that care about root slot accesses, but current we don't have any GC that does. 
We don't even have an API for it, although J9STATIC_OBJECT_STORE/LOAD could be 'abused' with providing class value NULL, but if we ever start doing it, we should really have dedicated APIs like J9STRONGROOT_OBJECT_STORE/LOAD

We require object access barriers for:
- heap objects slots, including their native struct extension that do not reside in heap. Hence we have J9STATIC_OBJECT_STORE/LOAD for class object slots extensions, aka class structs. 
- reading reference slots for native struct that act as _weak_ root (specifically, Monitor and String tables), and API is J9WEAKROOT_OBJECT_LOAD

In short, this line looks ok to me.

Reply:
Access to reference fields in VirtualThread native struct (Continuation or whatever), since they are not roots would require barrier, though.